WalletWasabi.Standard/CoinJoin/Coordinator/Rounds/CoordinatorRound.cs

```diff
@@ -1344,6 +1357,7 @@ public int RemoveAlicesBy(params Guid[] ids)
 				foreach (var id in ids)
 				{
 					numberOfRemovedAlices = Alices.RemoveAll(x => x.UniqueId == id);
+					UnconfirmedPeersAmount += numberOfRemovedAlices;
```

Comment:
Seems like this method is prone to gaming. Client Mallory could queue, dequeue, then queue again **even with the same input** and this number would go up. Can we do better based on the information in Alice? Maybe keep a running list of this round's previously registered inputs?

I know we spoke how this is a naive approach to a problem that's naive to solve in the first place. What do you see the next step to mitigate some bad behavior? Whatever the mechanism, it probably makes sense to have this pr anticipate what we'll use going forward if only as an exercise

Reply:
Yeah, that's a good point. I think having a queued list instead of a running number should fix this. We could check if Alice is currently queued when confirming status, if so we can remove Alice from the queued list.